### PR TITLE
Refactored the template configuration to a common parent and introduced Maven archetype

### PR DESCRIPTION
**Pull request by chrisdutz**
- Refactored the template configuration to a common parent
- Moved template static resources to a resource module
- Changed the original template into a maven archetype

### CODE REVIEW

File: tools/content-parent-pom/pom.xml [hunk 1]
```diff
@@ -196,16 +196,76 @@
...
+                    <execution>
+                        <id>copy-training-asciidoc-resources</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>unpack</goal>
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.apache.training</groupId>
```
Comment (RyanSkraba) [inline tools/content-parent-pom/pom.xml]: It looks like `content-master-resources` is built *after* `content-parent-pom` if you run mvn install from the root directory of the project.

I'm not sure this is intended!  Should `content-parent-pom` add a dependency on `content-master-resource`?

Comment (chrisdutz) [reply tools/content-parent-pom/pom.xml]: I'll look into this ...

Comment (chrisdutz) [reply tools/content-parent-pom/pom.xml]: Ok ... today I took some time to update this PR and @RyanSkraba you were right ... I added a provided scoped dependency to the pom ... now it should be built before the pom.

Now I would again wait about a day and merge if there are no objections (As there were none before, I guess this is ok)

File: tools/content-parent-pom/pom.xml [hunk 2]
```diff
@@ -75,7 +85,7 @@
                 <configuration>
                     <consoleOutput>true</consoleOutput>
                     <excludes>
-                        <!-- TODO: This is a mofified version of an asciidoctor file ... nor sure we can stick an Apache Header on it -->
+                        <!-- TODO: This is a modified version of an asciidoctor file ... nor sure we can stick an Apache Header on it -->
```
[the remark this answers is not in the document]
Comment (chrisdutz) [reply tools/content-parent-pom/pom.xml]: I'm pretty sure we can't stick it on it as it's only a super minor change.

**Pull request merged**